src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            addIfFirstNode(newNode);
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexValidation(index, size + 1);
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            addIfFirstNode(new Node<>(value));
+        } else if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+        } else if (index == size) {
+            add(value);
+            size--;
```

Comment:
why did you reduce the size here?  It looks like a hardcode. Let’s  think about how you can refactor your code to avoid it